connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala

```diff
@@ -120,33 +120,41 @@ class SparkConnectService(debug: Boolean)
       opType: String,
       observer: StreamObserver[V],
       userId: String,
-      sessionId: String): PartialFunction[Throwable, Unit] = {
+      sessionId: String,
+      events: Option[RequestEvents] = None): PartialFunction[Throwable, Unit] = {
     val session =
       SparkConnectService
         .getOrCreateIsolatedSession(userId, sessionId)
         .session
     val stackTraceEnabled = session.conf.get(PYSPARK_JVM_STACKTRACE_ENABLED)
 
-    {
```

Comment:
nit: instead of saving `val partial` and then having `partial.andThen`, would
```
{

}.andThen(...)
```
work?

Reply:
Unfortunately does not work.
I do see the option of doing
```
PartialFunction[Throwable, (Throwable, Throwable)] {
}.andThen
```
but that's deprecated and recommend using original